python/GafferUI/ScriptWindow.py

```diff
@@ -90,6 +98,19 @@ def setLayout( self, compoundEditor ) :
 		assert( compoundEditor.scriptNode().isSame( self.scriptNode() ) )
 		self.__listContainer.append( compoundEditor, expand=True )
 
+		if len( self.__menuContainer ) > 1 :
+			del self.__menuContainer[1:]
+
+		if hasattr( compoundEditor, "settings" ) :
```

Comment:
This check seems redundant - the `settings()` method is provided by the Editor base class, so should always be available.

Reply:
Yep, it is indeed. Removed it and another redundant check in a823a614f40c5e3e30267c0e04f0ddcdfe51fe2a